homeassistant/components/rest/__init__.py

```diff
@@ -1,4 +1,170 @@
 """The rest component."""
 
-DOMAIN = "rest"
+import asyncio
+import logging
+
+import httpx
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_HEADERS,
+    CONF_METHOD,
+    CONF_PARAMS,
+    CONF_PASSWORD,
+    CONF_PAYLOAD,
+    CONF_RESOURCE,
+    CONF_RESOURCE_TEMPLATE,
+    CONF_SCAN_INTERVAL,
+    CONF_TIMEOUT,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_DIGEST_AUTHENTICATION,
+    SERVICE_RELOAD,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity_component import (
+    DEFAULT_SCAN_INTERVAL,
+    EntityComponent,
+)
+from homeassistant.helpers.reload import async_reload_integration_platforms
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import COORDINATOR, DOMAIN, PLATFORM_IDX, REST, REST_IDX
+from .data import RestData
+from .schema import CONFIG_SCHEMA  # noqa:F401 pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
 PLATFORMS = ["binary_sensor", "notify", "sensor", "switch"]
+COORDINATOR_AWARE_PLATFORMS = [SENSOR_DOMAIN, BINARY_SENSOR_DOMAIN]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the rest platforms."""
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
```

Comment:
It's a bit silly to use an entity component just for the `async_prepare_reload` function. We should probably extract that function from that helper into a standalone function.

Reply:
(We can do that in a follow-up PR)

Reply:
queued for followup